### DISCUSSION TITLE
App Dir - How do you get URL slug from layout.tsx

### DISCUSSION THREAD

**Discussion by eliu-fracta** [vercel/next.js#47481] [Help]
### Summary

For NEXT13 App Dir, I know that within page.tsx, you can use params prop to get URL slugs. However, how can I get URL slugs from layout.tsx?

**Comment by la-flor** [comment]
The same way. 

``` Javascript
export default async function Layout({
  children,
  params,
}: {
  children: React.ReactNode;
  params: { id: string };
}) {
  const product = await getProduct(params.id);

  return <div>{product.name}</div>
}
```

**Reply by edward1127** [reply]
thank you!

**Reply by la-flor** [reply]
No problem!  Yeah, I feel like the documentation could have been a little clearer on its use in both cases. 🤷
